sqlglot/dataframe/sql/dataframe.py

```diff
@@ -420,59 +427,84 @@ def join(
         **kwargs,
     ) -> DataFrame:
         other_df = other_df._convert_leaf_to_cte()
-        pre_join_self_latest_cte_name = self.latest_cte_name
-        columns = self._ensure_and_normalize_cols(on)
-        join_type = how.replace("_", " ")
-        if isinstance(columns[0].expression, exp.Column):
-            join_columns = [
-                Column(x).set_table_name(pre_join_self_latest_cte_name) for x in columns
+        join_columns = self._ensure_list_of_columns(on)
+        # We will determine actual join expression later so we provide a dummy one for now.
+        join_expression = self.expression.copy().join(
+            other_df.latest_cte_name, on=exp.Column(), join_type=how.replace("_", " ")
```

Comment:
why are doing a null column? that's invalid

Reply:
I explain that in the comment above. This on expression can be anything since we calculate it on later steps

Reply:
why don't you just leave it null?

Reply:
Ok that works.